server.c

```diff
@@ -19,32 +21,44 @@ struct request {
     char query[2048];
 };
 
-void request_parse(struct request* req, const char* data) {
+#define RESPONSE_TEMPLATE_MACRO "HTTP/1.1 200 OK\r\n"\
+        "Server: yeet/1.0\r\n"\
+        "Content-Type: text/plain\r\n\r\n"
+/* We don't want to actually save the null byte */
+char RESPONSE_TEMPLATE[sizeof(RESPONSE_TEMPLATE_MACRO) - 1] = RESPONSE_TEMPLATE_MACRO;
```

Comment:
@mxmeinhold because I wanted to be able to store it as a char array without the null byte at the end. I would've set it directly, but I needed to be able to do the `sizeof()` check.

Reply:
Why do you need to avoid the null byte? It looks like below you're doing memcpy with sizeof, why not pre store the size, and skip the sizeof call? If you use (potentially precomputed) strlen, it won't include the null byte when you call memcpy.

Reply:
@mxmeinhold sizeof isn't a real call, it's done at compile-time. This saves a byte of program memory and means we don't need to do the subtraction at runtime.